smarts/core/smarts.py

```diff
@@ -303,6 +310,11 @@ def reset(self, scenario: Scenario):
     def setup(self, scenario: Scenario):
         self._scenario = scenario
 
+        self._use_realtime_clock = scenario.traffic_history and (
+            not self._traffic_sim.headless or not self._envision.headless
+        )
```

Comment:
I see why the real-time option might be useful for imitation learning but I have a few concerns. My first inclination here is that it should not be attached to the traffic history because there are other uses for a realtime clock option and this binds the option to the traffic history. 

Secondly, Envision is already supposed to play back at real-time and I believe this complicates fixing Envision so I do not think Envision should be included with this option.

Otherwise this is useful to `sumo-gui` but there is another alternative. It is possible to send the `--gui-settings-file` option to `sumo-gui` to add step delay and breakpoints via a configuration file: https://sumo.dlr.de/docs/sumo-gui.html#configuration_files.

Reply:
Re the traffic-history limitation, I agree with you.  I just was hesitant to auto-set it and change the behavior for other things that might have inadvertently set `headless` to False but where no one watches the gui/envision.

Re Envision, there is still a bug there, but you're right that this "masks" the problem.  

I will look into the `--gui-settings-file` option...

Reply:
Using step delay via a `--gui-settings-file` could work assuming we know (approximately) how long our steps really take.  But if these will be variable, or if we'll be improving our step time, then we may have to tweak this occasionally.

I'm willing to do this, but let's chat about it on Monday first.  I do see other advantages to having a near-real-time mode (in particular, asyncrhonously running alongside ROS nodes), but there may be better ways to achieve that, so I can be persuaded to leave this out or now (and/or use the sumo-gui delay setting).